pkg/endpoint/bpf.go

```diff
@@ -1017,10 +1020,10 @@ func (e *Endpoint) deletePolicyKey(keyToDelete policy.Key, incremental bool, had
 		return false
 	}
 
-	if hadProxy != nil {
-		if entry, ok := e.realizedPolicy.PolicyMapState[keyToDelete]; ok && entry.ProxyPort != 0 {
-			*hadProxy = true
-		}
+	var entry policy.MapStateEntry
+	var ok bool
+	if entry, ok = e.realizedPolicy.PolicyMapState[keyToDelete]; ok && entry.ProxyPort != 0 && hadProxy != nil {
```

Comment:
As far as I can tell, Go does not guarantee the ordering of resolving the independent conditions in this statement. The `entry.ProxyPort` check must be split onto another line.

Following this Go playground, I believe that `*hadProxy` will never be updated:

https://play.golang.org/p/HqckF5pYktm

Reply:
@joestringer I don't follow your statement. In your Go playground link, you were returning `err` as `nil` so it would never try to check `res`? I printed `res`, and it is indeed `true` (the value returned by the function).

Reply:
Good point, glad I had an example to dispel :sweat_smile: 

That is at least consistent with the statements on this website around evaluation of conditional statements (not official go page, but seems authoritative): https://go101.org/article/evaluation-orders.html